apisix/plugins/log-rotate.lua

```diff
@@ -143,10 +157,18 @@ local function rename_file(log, date_str)
         core.log.info("file exist: ", new_file)
         return new_file
     end
-
-    local ok, err = os_rename(log.file, new_file)
+    local filename = log.file
+    -- create the file if it does not exist
+    local exists = lfs.attributes(filename, "mode") == "file"
+    if not exists then
+        local file = io_open(filename, "w")
+        if file then
+            file:close()
+        end
+    end
```

Comment:
Why do you create a new file here, i think it's better to record a error log and exit the function

Reply:
@Revolyssup please check this

Reply:
Is it necessay to add this check?
The original logic looked fine,  `os.rename` would log when it failed and then skip

Reply:
@monkeyDluffy6017 removed this check